### ISSUE TITLE
unable to update to beta 8 under windows

### ISSUE THREAD

**Issue by lucamorelli** [angular/angular-cli#1233]
Hi, I'm trying to update from beta 6 to 8 but I receive this error

```
npm ERR! enoent ENOENT: no such file or directory, rename 'C:\Users\luca\AppData\Roaming\npm\node_modules\.staging\abbre
v-bd79b7a1' -> 'C:\Users\luca\AppData\Roaming\npm\node_modules\angular-cli\node_modules\ember-cli\node_modules\npm\node_
modules\abbrev'

```
I suppose the problema is the length of the  path. How can I solve this?

**Comment by anode7** [comment]
`npm r -g angular-cli` and `npm i -g angular-cli` works fine...

**Comment by lucamorelli** [comment]
under windows? I know the commands, but in my windows 10 machine the installation fails for the lenght of the path

**Comment by mark-zacharias** [comment]
I was having the same issue but following the odd update instructions worked for me.
[https://github.com/angular/angular-cli#updating-angular-cli](https://github.com/angular/angular-cli#updating-angular-cli)
```
npm uninstall -g angular-cli
npm cache clean
npm install -g angular-cli@lates
```

Not sure why I need to uninstall and clean the cache but oh well.

**Comment by burhan** [comment]
I don't know of a solution to this problem, but you can change the location of the default `node_modules` directory to path with fewer depths; for example, I have mine on `D:\node_modules`.  This is used when installing globally `-g`, local installs will have to go into the project folder's `node_modules` directory, and then you'll run into the same path-length problems.

Unfortunately you cannot rename the directory, it has to be called `node_modules`.

To summarize:

1. Change the global `node_modules` location. If you do this, make sure you update your `PATH` variable as well. See `npm help folders` for more.

2. Start your project in a directory closest to the root. I usually put everything in `D:\dev`

Here's a tip - if you set the directory to a location at `C:\`, you'll need to manually adjust the permissions as modern versions of Windows restrict permissions for folders that are rooted at `C:\`.

**Comment by kylecordes** [comment]
I have not experienced this path length issue myself, even on windows, in a long time. The last I saw, this problem was fundamentally addressed when NPM was enhanced to produce a maximally flat node modules structure.

I don't see anything in this thread about node versions. I think there is a good chance the issue here would be addressed by upgrading to the current node (as stated in the angular documentation quick start page: https://angular.io/docs/ts/latest/quickstart.html  "Verify that you are running at least node v5.x.x and npm 3.x.x").

If this turns out to be the cause of this issue under discussion here, maybe we can also treat this item as (yet another?) feature request that CLI check at start up that it is running under a sufficiently new version of node, and halt with an error directing the user to upgrade node if it is not.

Regardless of whether this turns out to be a node version problem, I strongly recommend against widely directing CLI users to reconfigure their system to store node modules at the root of their C/D drive. This would be another strong negative quality signal, as countless other node-based systems are used by many people (a million? millions?) across our industry without doing so. A properly written node program on a sufficiently new node will "just work" without any of these machinations.

**Comment by lucamorelli** [comment]
updated to 6.2.2
uninstalled
reinstalled
now seems ok.
thanks
